src/TransitionSystems/location_id.rs

```diff
@@ -7,11 +7,35 @@ pub enum LocationID {
     Conjunction(Box<LocationID>, Box<LocationID>),
     Composition(Box<LocationID>, Box<LocationID>),
     Quotient(Box<LocationID>, Box<LocationID>),
-    Simple(String),
+    Simple(SimpleID),
     /// Used for representing a partial state and it is generated when a location's name is set as `_`
     AnyLocation(),
 }
 
+#[derive(Debug, Clone, Eq, Hash, PartialEq)]
+pub struct SimpleID {
+    location_id: String,
+    component_id: Option<String>,
+}
+
+impl SimpleID {
+    pub fn new(location_id: String, component_id: Option<String>) -> Self {
+        Self {
+            location_id,
+            component_id,
+        }
+    }
+
+    #[allow(dead_code)]
+    fn component_id(&self) -> Option<&String> {
+        self.component_id.as_ref()
+    }
```

Comment:
If it is not used, I think it should be removed. If it will be used in the future, add a comment stating that (perhaps a TODO)

Reply:
fixed